sdk/src/Services/CognitoIdentity/Custom/CognitoAWSCredentials.cs

```diff
@@ -443,14 +444,14 @@ private bool ShouldRetry(AmazonCognitoIdentityException e)
         {
             add
             {
-                lock (this)
+                lock (identityChangeEventLock)
```

Comment:
Why does this lock exist at all since it is only setting and getting a single member variable and usage doesn't seem to be protected with the same lock elsewhere. Is there someplace else where `mIdentityChangedEvent` is used that should be locked with `identityChangeEventLock`?

Reply:
It is not setting but adding a value to a collection via the `+=` and `-=` operators. If memory services me correct we had an issue a long time ago with adding handlers to events for Unity or Xamarin. So this was probably done in defensive of that issue. It quite likely isn't needed anymore but this is not a high throughput area of code and I don't want make that change now.